src/clib/calculate_gamma.c

```diff
@@ -68,39 +75,48 @@ int local_calculate_gamma(chemistry_data *my_chemistry,
     double x, nH2, number_density, GammaH2Inverse, 
       GammaInverse = 1 / (my_chemistry->Gamma - 1.0);
 
+    /* parallelize the k and j loops with OpenMP
+     * (these loops are flattened them for better parallelism) */
 #   ifdef _OPENMP
 #   pragma omp parallel for schedule( runtime ) \
-    private( x, nH2, number_density, GammaH2Inverse )
+    private( outer_ind, index, x, nH2, number_density, GammaH2Inverse )
 #   endif
-    for (i = 0; i < size; i++) {
- 
-      /* Compute relative number abundence of molecular hydrogen. */
- 
-      number_density =
-        0.25 * (my_fields->HeI_density[i] + my_fields->HeII_density[i] +
-                my_fields->HeIII_density[i]) +
-        my_fields->HI_density[i] + my_fields->HII_density[i] +
-        my_fields->HM_density[i] + my_fields->e_density[i];
+    for (outer_ind = 0; outer_ind < ind_helper.outer_ind_size; outer_ind++){
+
+      const grackle_index_range range = _inner_range(outer_ind, &ind_helper);
```

Comment:
Does this need to be in the private list? I guess not since it's declared in the loop, but just checking.

Reply:
I can confirm that since this is declared in the loop, it doesn't need to be in the private list. I confirmed this when I wrote the code.

I just tried to look for confirmation of this again:
- This [linked section](https://www.openmp.org/spec-html/5.1/openmpsu113.html#x148-1610002.21.1.1) of the current verision of the OpenMP specification confirms this behavior
- This is also confirmed in version 1.0 of the specification ([linked here](https://www.openmp.org/wp-content/uploads/cspec10.pdf)). On page 2 of the specification, while defining the meaning of "private", they make the following statement (the emphasis is mine):
  > Note that there are several ways to specify that a variable is private: **a definition within the parallel region**, a `threadprivate` directive, a `private`, `firstprivate`, `lastprivate`, or `reduction `clause, or use of the variable as a loop control variable.

Reply:
Thanks for the thorough check on this. I think we can merge this now. Great addition!